--- a/Source\\RestRequest.h
+++ b/Source\\RestRequest.h
@@ -38,12 +38,19 @@
         if (hasFields)
         {
             MemoryOutputStream output;
-
-            fields.writeAsJSON (output, 0, false, 20);
+            fields.writeAsJSON (output, juce::JSON::FormatOptions{}.withIndentLevel(0).withMaxDecimalPlaces(20));
             urlRequest = urlRequest.withPOSTData (output.toString());
         }
 
-        std::unique_ptr<InputStream> input (urlRequest.createInputStream (hasFields, nullptr, nullptr, stringPairArrayToHeaderString(headers), 0, &response.headers, &response.status, 5, verb));
+        std::unique_ptr<InputStream> input = urlRequest.createInputStream (
+            juce::URL::InputStreamOptions(hasFields ? juce::URL::ParameterHandling::inPostData : juce::URL::ParameterHandling::inAddress)
+                .withExtraHeaders(stringPairArrayToHeaderString(headers))
+                .withConnectionTimeoutMs(0)
+                .withResponseHeaders(&response.headers)
+                .withStatusCode(&response.status)
+                .withNumRedirectsToFollow(5)
+                .withHttpRequestCmd(verb)
+        );
 
         response.result = checkInputStream (input);
         if (response.result.failed()) return response;
